src/components/Form/Combobox.js

```diff
@@ -0,0 +1,333 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import memoize from 'lodash/memoize';
+
+import ComboboxDropdown from './ComboboxDropdown';
+import ComboboxDropdownList from './ComboboxDropdownList';
+import ComboboxDropdownListItem from './ComboboxDropdownListItem';
+
+import {
+  ClickOutside,
+  Icon,
+  InputRaw,
+  Listbox,
+  Pill,
+} from '../..';
+
+export const propTypes = {
+  /**
+   * class name
+   */
+  className: PropTypes.string,
+  /**
+   * Whether the picklist closes automatically on an outside click
+   */
+  closeOnClickOutside: PropTypes.bool,
+  /**
+   * close the combobox when selecting an item
+   */
+  closeOnSelect: PropTypes.bool,
+  /**
+   * sets the number of items being displayed
+   */
+  error: PropTypes.string,
+  /**
+   * sets the number of items being displayed
+   */
+  height: PropTypes.number,
+  /**
+   * unique id
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * whether the input is disabled
+   */
+  disabled: PropTypes.bool,
+  /**
+   * whether the input lable is visible
+   */
+  hideLabel: PropTypes.bool,
+  /**
+   * toggle inline listbox mode
+   */
+  inlineListbox: PropTypes.bool,
+  /**
+   * list of displayed items
+   * `{ key: 'id123', label: 'first option', selected: false }`
+   */
+  items: PropTypes.arrayOf(PropTypes.shape({
+    key: PropTypes.any,
+    label: PropTypes.string,
+    selected: PropTypes.bool,
+  })),
+  /**
+   * label for input label
+   */
+  labelInput: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.node,
+  ]),
+  /**
+   * label for multiple selected items
+   */
+  labelMultiselect: PropTypes.string,
+  /**
+   * called when adding a custom item in case the combobox isnt in readOnly mode
+   */
+  onAdd: PropTypes.func,
+  /**
+   * callback for changes in the input field
+   */
+  onChange: PropTypes.func,
+  /**
+   * triggered whenever an item was clicked, has the item's key as parameter
+   */
+  onSelect: PropTypes.func.isRequired,
+  /**
+   * placeholder for the input. if a selection is present,
+   * you should indicate it
+   */
+  placeholder: PropTypes.string.isRequired,
+  /**
+   * toggle readonly mode
+   */
+  readOnly: PropTypes.bool,
+  /**
+   * whether the combobox is required
+   */
+  required: PropTypes.bool,
+  /**
+   * Picklist sizes: small, medium, large
+   */
+  size: PropTypes.oneOf(['small', 'medium', 'large']),
+  /**
+   * value of the input element
+   */
+  value: PropTypes.string,
+};
+
+export class ComboboxRaw extends Component {
+  static propTypes = propTypes
+
+  static defaultProps = {
+    className: null,
+    closeOnClickOutside: false,
+    closeOnSelect: true,
+    error: null,
+    height: null,
```

Comment:
I would set a default height e.g. `5` or `7` as we always run into problems with long dropdowns.